packages/gatsby/src/redux/persist.ts

```diff
@@ -0,0 +1,11 @@
+import v8 from "v8"
+import { readFileSync, writeFile } from "fs-extra"
+import { IReduxState } from "./types"
+
+const file = (): string => `${process.cwd()}/.cache/redux.state`
+
+export const readFromCache = (): IReduxState =>
+  v8.deserialize(readFileSync(file()))
+
+export const writeToCache = (contents: IReduxState): Promise<void> =>
+  writeFile(file(), v8.serialize(contents))
```

Comment:
This changes it from sync to async version of `writeFile` - it was actually pretty intentional to use `sync` version here (as a work around problems with periodically persisting state - with async and us saturating event loop pretty heavily, async file writing was taking so long, that sometimes (in case of state for larger sites), it would took longer to write file than our debounce time for persisting state trigger - so we were in some cases doing lot of concurrent writes to same file, causing very weird results.

Ideally, we could fix our scheduler for that, but for now - please let's keep it 1:1

Reply:
Thank you for sharing that! I changed it because I noticed some upstream calls were calling it with `await` which made me assume it was supposed to be async. I'll undo this <3